python/gstaichi/lang/ast/ast_transformers/call_transformer.py

```diff
@@ -257,11 +257,17 @@ def build_Call(ctx: ASTTransformerContext, node: ast.Call, build_stmt, build_stm
             # not related to dataclasses). Necessary for calling further child functions
             build_stmts(ctx, node.args)
             build_stmts(ctx, node.keywords)
+
             node.args = CallTransformer._expand_Call_dataclass_args(node.args)
             node.keywords = CallTransformer._expand_Call_dataclass_kwargs(node.keywords)
+
             # create variables for the now-expanded dataclass members
-            build_stmts(ctx, node.args)
-            build_stmts(ctx, node.keywords)
+            for arg in node.args:
+                if not hasattr(arg, "ptr"):
+                    build_stmt(ctx, arg)
+            for arg in node.keywords:
+                if not hasattr(arg, "ptr"):
+                    build_stmt(ctx, arg)
```

Comment:
Minor naming suggestion: `for keyword in node.keywords:` in stead of  `for arg in node.keywords`.

Second, the solution seem a bit hacky, as now we are calling similar functions multiple times. 

```
build_stmts(ctx, node.args)
build_stmts(ctx, node.keywords)
build_stmt(ctx, arg)
build_stmt(ctx, keyword)
```

not sure, but how about?
```
new_args, new_keywords = expand_call_dataclass(node.args, node.keywords)

build_stmts(ctx, new_args)
build_stmts(ctx, new_keywords)

node.args = new_args
node.keywords = new_keywords
```

Reply:
addressed in [bd51400](https://github.com/Genesis-Embodied-AI/gstaichi/pull/228/commits/bd5140092894b8f08ba37da3859c37e5f4477566)